include/private/gcconfig.h

```diff
@@ -2890,7 +2890,8 @@ EXTERN_C_BEGIN
 # ifdef RISCV
 #   define MACH_TYPE "RISC-V"
 #   define CPP_WORDSZ __riscv_xlen /* 32 or 64 */
-#   define ALIGNMENT (CPP_WORDSZ/8)
+//#   define ALIGNMENT (CPP_WORDSZ/8)  Dejice original value for RISCV
+#   define ALIGNMENT ((CPP_WORDSZ<<1)/8)  // Dejice changed for cheri-riscv
```

Comment:
Aha, so this is a case where "RISC-V" alone isn't a sufficient discriminator: we need to say `if RISCV && CHERI` I think? Is there a `#define` in the CHERI headers that we can use for these purposes.

Reply:
fixed in commit [408f667b](https://github.com/capablevms/bdwgc/pull/1/commits/408f667bb59d4d389e44ff70d5631bf922413ed9)